proof/mint.go

```diff
@@ -259,21 +263,28 @@ func committedProofs(baseProof *Proof, taprootAssetRoot *commitment.TapCommitmen
 		// transition proofs.
 		assetProof.GenesisReveal = &newAsset.Genesis
 
-		// We also need to reveal the group key.
+		// If the asset has a group key, we only need to populate the
+		// group key reveal if the asset is the group anchor.
 		if newAsset.GroupKey != nil {
 			groupKey := newAsset.GroupKey
 
-			// All zero tapscript root means there is none.
-			var tapscriptRoot []byte
-			if groupKey.TapscriptRoot != [32]byte{} {
-				tapscriptRoot = groupKey.TapscriptRoot[:]
-			}
-
-			assetProof.GroupKeyReveal = &asset.GroupKeyReveal{
-				RawKey: asset.ToSerialized(
-					groupKey.RawKey.PubKey,
-				),
-				TapscriptRoot: tapscriptRoot,
+			err := anchorVerifier(
+				&newAsset.Genesis, groupKey,
+			)
+			if err == nil {
```

Comment:
I think adjusting the naming of `anchorVerifier` and also the callback type could make this easier to understand. 

IIUC, we're just checking here that this asset is the actual anchor asset for the group.